qiling/os/posix/syscall/unistd.py

```diff
@@ -432,8 +432,18 @@ def __read_str_array(addr: int) -> Iterator[str]:
     if ql.code:
         return
 
-    ql._uc = ql.arch.init_uc
-    QlCoreHooks.__init__(ql, ql._uc)
+    # recreate cached uc
+    del ql.arch.uc
```

Comment:
Why uc is re-generated here?

Reply:
The original code did that on `execve` not entirely sure why (perhaps that was a quick way to discard everything and start fresh..?). I just followed it.

Reply:
But the two code behaves differently if `ql._uc` and `ql.arch.init_ic` points to the same object.

Reply:
I'm not sure that I follow you here.. The old code regenerated a new Uc instance by calling `init_uc` (which actually was used only here and on Qiling init), it wasn't a cached property.
The new code has Uc cached, and the `del` statement is how you force it to be re-created next time you retrieve it.
Both assign Qiling with a new Uc instance, it is just done a bit differently.